soroban-env-host/src/native_contract/token/contract.rs

```diff
@@ -38,64 +36,63 @@ pub trait TokenTrait {
     /// init creates a token contract that does not wrap an asset on the classic side.
     fn init(e: &Host, admin: Identifier, metadata: TokenMetadata) -> Result<(), HostError>;
 
-    fn nonce(e: &Host, id: Identifier) -> Result<BigInt, HostError>;
+    fn nonce(e: &Host, id: Identifier) -> Result<i128, HostError>;
 
-    fn allowance(e: &Host, from: Identifier, spender: Identifier) -> Result<BigInt, HostError>;
+    fn allowance(e: &Host, from: Identifier, spender: Identifier) -> Result<i128, HostError>;
 
     fn approve(
         e: &Host,
         from: Signature,
-        nonce: BigInt,
+        nonce: i128,
         spender: Identifier,
-        amount: BigInt,
+        amount: i128,
```

Comment:
What was the reasoning again for using `i128` instead of `u128` in the interface?

Reply:
"because some tokens might somehow want to deal with overdrafts and/or negative amounts"